### PR TITLE
Error handling, "try...catch"

### PR DESCRIPTION
**Pull request by mahdiHash**
Please review and comment/commit if anything is wrong.

### CODE REVIEW

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 1]
```diff
@@ -1,295 +1,295 @@
...
 try {
   {{{{{{{{{{{{
 } catch (err) {
-  alert("The engine can't understand this code, it's invalid");
+  alert("موتور جاوااسکریپت نمی‌تواند این کد را متوجه شود. این کد نامعتبر است.");
 }
 ```
 
-The JavaScript engine first reads the code, and then runs it. The errors that occur on the reading phase are called "parse-time" errors and are unrecoverable (from inside that code). That's because the engine can't understand the code.
+موتور جاوااسکریپت ابتدا کد را می‌خواند و سپس آن را اجرا می‌کند. ارورهایی که در فاز خواندن رخ می‌دهند، ارورهای «زمان تجزیه (parse-time errors)» نامیده می‌شوند و قابل پوشش نیستند (از درون همان کد). به این دلیل که موتور نمی‌تواند کد را متوجه شود.
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:88]: More attention here.

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 2]
```diff
@@ -1,295 +1,295 @@
...
-  // Can also show an error as a whole
-  // The error is converted to string as "name: message"
+  // می‌توانستیم ارور را به طور کامل هم نشان دهیم
+  // به رشته تبدیل می‌شود «name: message» ارور به صورت
   alert(err); // ReferenceError: lalala is not defined
 }
 ```
 
-## Optional "catch" binding
+## پیوند اختیاری «catch»
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:164]: More attention here.

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 3]
```diff
@@ -1,295 +1,295 @@
...
 
-For built-in errors (not for any objects, just for errors), the `name` property is exactly the name of the constructor. And `message` is taken from the argument.
+برای ارور‌های درون‌ساخت (نه برای هر شیءای، فقط برای ارورها)، ویژگی `name` دقیقا اسم سازنده است. و `message` از آرگومان گرفته می‌شود.
 
-For instance:
+برای مثال:
 
 ```js run
-let error = new Error("Things happen o_O");
+let error = new Error("اتفاقاتی رخ می‌دهد o_O");
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:286]: More attention here.

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 4]
```diff
@@ -328,44 +328,44 @@ try {
...
   // ...
 } catch (err) {
   alert("JSON Error: " + err); // JSON Error: ReferenceError: user is not defined
-  // (no JSON Error actually)
+  // (نیست JSON Error در واقع)
 }
 ```
 
-Of course, everything's possible! Programmers do make mistakes. Even in open-source utilities used by millions for decades -- suddenly a bug may be discovered that leads to terrible hacks.
+قطعا، هر چیزی ممکن است! برنامه‌نویسان حتما اشتباهاتی می‌کنند. حتی در تسهیلات متن‌باز (open-source) که برای ده‌ها سال میلیون‌ها بار استفاده شده‌اند -- ناگهان یک خطا یا باگ (bug) ممکن است کشف شود که منجر به هک‌های وحشتناک می‌شود.
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:354]: More attention here.

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 5]
```diff
@@ -441,70 +441,70 @@ try {
...
+2. اگر شما «نه» بگویید، سپس `try -> finally`.
 
-The `finally` clause is often used when we start doing something and want to finalize it in any case of outcome.
+بند `finally` اغلب زمانی استفاده می‌شود که ما انجام کاری را شروع می‌کنیم و می‌خواهیم با هر نتیجه‌ای آن را به پایان برسانیم.
 
-For instance, we want to measure the time that a Fibonacci numbers function `fib(n)` takes. Naturally, we can start measuring before it runs and finish afterwards. But what if there's an error during the function call? In particular, the implementation of `fib(n)` in the code below returns an error for negative or non-integer numbers.
+برای مثال، ما می‌خواهیم زمانی که یک تابع اعداد فیبوناچی `fib(n)` می‌گیرد را محاسبه کنیم. طبیعتا، ما می‌توانیم قبل از اینکه اجرا شود اندازه‌گیری را آغاز کنیم و سپس آن را تمام کنیم. اما اگر در حین فراخوانی تابع ارور ایجاد شود چه؟ به خصوص در کد پایین، پیاده‌سازی `fib(n)` به ازای اعداد منفی یا غیر صحیح یک ارور برمی‌گرداند.
 
-The `finally` clause is a great place to finish the measurements no matter what.
+بند `finally` مکانی عالی برای اتمام اندازه‌گیری‌ها است؛ هر چه می‌خواهد رخ دهد.
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:496]: More attention here.

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 6]
```diff
@@ -623,53 +623,53 @@ For instance:
 */!*
 
   function readData() {
-    badFunc(); // Whoops, something went wrong!
+    badFunc(); // !اوه، یک جای کار می‌لنگد
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:626]: More attention here.

File: 1-js/10-error-handling/1-try-catch/article.md [hunk 7]
```diff
@@ -623,53 +623,53 @@ For instance:
...
 
-Error objects have following properties:
+شیءهای ارور ویژگی‌های پایین را دارند:
 
-- `message` -- the human-readable error message.
-- `name` -- the string with error name (error constructor name).
-- `stack` (non-standard, but well-supported) -- the stack at the moment of error creation.
+- `message` -- پیام ارور که برای انسان قابل خواندن است.
+- `name` -- رشته حاوی اسم ارور (اسم تابع سازنده ارور)
+- `stack` (استاندارد نیست، اما به خوبی پشتیبانی می‌شود) -- پشته‌ای که در لحظه ایجاد ارور وجود دارد.
```
Comment (mahdiHash) [inline 1-js/10-error-handling/1-try-catch/article.md:667]: More attention here, too.

File: 1-js/10-error-handling/1-try-catch/article.md (hunk 1 above)
Comment (mahdyar) [reply 1-js/10-error-handling/1-try-catch/article.md:88]: LGTM!

File: 1-js/10-error-handling/1-try-catch/article.md (hunk 4 above)
Comment (mahdyar) [reply 1-js/10-error-handling/1-try-catch/article.md:354]:
```suggestion
قطعا، هر چیزی ممکن است! برنامه‌نویسان حتما اشتباهاتی می‌کنند. حتی در تسهیلات متن‌باز (open-source) که برای ده‌ها سال میلیون‌ها بار استفاده شده‌اند -- ناگهان یک خطا یا باگ (bug) ممکن است کشف شود که منجر به رخنه‌های وحشتناک می‌شود.
```

File: 1-js/10-error-handling/1-try-catch/article.md (hunk 5 above)
Comment (mahdyar) [reply 1-js/10-error-handling/1-try-catch/article.md:496]:
```suggestion
بند `finally` مکانی عالی برای اتمام اندازه‌گیری‌ها است؛ هر اتفاقی که بیوفتد.
```

**Review by mahdyar — CHANGES_REQUESTED** [review]
Please review the suggested change(s) and request another review.

**Comment by mahdiHash** [comment]
Thanks for your review, @mahdyar.
/done

File: 1-js/10-error-handling/1-try-catch/article.md (hunk 6 above)
Comment (Beroozam) [reply 1-js/10-error-handling/1-try-catch/article.md:626]:
```suggestion
    badFunc(); // !اوه، یه جای کار می‌لنگه
```

**Review by mahdyar — APPROVED** [review]

**Comment by mahdyar** [comment]
Thank you, @mahdiHash!

File: 1-js/10-error-handling/1-try-catch/article.md (hunk 6 above)
Comment (mahdiHash) [reply 1-js/10-error-handling/1-try-catch/article.md:626]: > 

I think the formal form is better.  
Thanks for your suggestion, though.

**Pull request merged**